New feature: add CORS support

etiktin opened the pull request:
When a developer is working on a website/webapp that is accessing miniProxy using AJAX, the developer might have issues with CORS (Cross-origin resource sharing) that will prevent him/her from getting
the proxified data. For example if the webapp and the service are not on the same domain, the browser will block the request.
I added a way to easily enable CORS support. All the developer needs to do is edit config.php and set $forceCORS to true (the default is false, which will act exactly as before).

Using this will make it a lot easier to use the service through AJAX.

etiktin commented:
@joshdick, I moved the configuration inline like you suggested.

joshdick commented:
This code doesn't handle the case where the server response also contains its own set of CORS headers, so if CORS is enabled, both the proxy's headers and the original headers will be served together, which will almost certainly cause issues and confusion.

I'm not sure how to fix this besides adding something like `Access-Control-.*` to `$header_blacklist_pattern` when CORS is enabled, but that's not a great solution either.

Thoughts?

etiktin commented:
See my last commit. I moved the enableCORS() call to after adding all the response headers we get from cURL. The default behavior of the header() function is to overwrite similar headers, so even if the response already had CORS set to some origin, we will overwrite it.

I also changed the styling to better fit the current style.

etiktin commented:
Josh, did you see my last comment?

joshdick commented:
I've merged this in via 9c36ff5cd61e5499bcf6301550e6f106c04b3236, which closes this pull request.

Thanks!
